tests/datasets/test_synthetic_lm_data.py

```diff
@@ -0,0 +1,117 @@
+from itertools import product
```

Comment:
This isn't standard practice, but I wonder if we should start annotating our tests a bit more. Their intent can often be inferred from the code, but it might make things easier for everyone if their intent was clearly stated by the author.